cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterUtils.java

```diff
@@ -1058,6 +1074,12 @@ static boolean fetchCompletedTask(HttpServletRequest request) {
     return getBooleanParam(request, FETCH_COMPLETED_TASK_PARAM, false);
   }
 
+  @Nullable
+  private static Long parseParamAsLong(HttpServletRequest request, String paramName) {
+    String parameterString = caseSensitiveParameterName(request.getParameterMap(), paramName);
+    return parameterString == null ? null : Long.parseLong(request.getParameter(parameterString));
+  }
+
```

Comment:
Cool. I added the below method:

```
  public static Long getLongParam(HttpServletRequest request, String parameter, @Nullable Long defaultIfMissing) {
    String parameterString = caseSensitiveParameterName(request.getParameterMap(), parameter);

    if (parameterString == null) {
      return defaultIfMissing;
    }
    return Long.parseLong(request.getParameter(parameterString));
  }
```

Note that the below statement does not work since `defaultIfMissing` is nullable and the implicit unboxing process might get NPE

`return parameterString == null ? defaultIfMissing : Long.parseLong...`

Reply:
More specifically, when we use `condition ? a : b`, it tries to find the common denominator between a type and b type. In the below case, the common type denominator is primitive `long`. That means it will try to unbox `defaultIfMissing` which could be null in this case.

`parameterString == null ? defaultIfMissing : Long.parseLong(...)`

Reply:
@Lincong Thanks for the explanation.
I guess what you are looking for is the following:
```
return parameterString == null ? defaultIfMissing : Long.valueOf(request.getParameter(parameterString));
```

This has the benefit of 
1. not doing the extra boxing to return Long value of parsed long, and
2. being more concise

Reply:
Right. This one works since `Long.valueOf(...)` return `Long` instead of `long`. So `defaultIfMissing` does not get unboxed